modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/SelectWithOption.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import DropDown from '@clayui/drop-down';
+import ClayIcon from '@clayui/icon';
+import classNames from 'classnames';
+import React, {useRef, useState} from 'react';
+
+import './index.scss';
+import {FieldBase} from '../FieldBase';
+
+type Item = {
+	label: string;
+	options: LabelValueObject[];
+	type: string;
+};
+interface SelectWithOptionProps
+	extends React.SelectHTMLAttributes<HTMLSelectElement> {
+	ariaLabel?: string;
+	disabled?: boolean;
+	error?: string;
+	feedbackMessage?: string;
+	items: Item[];
+	label?: string;
+	onSelectChange: (label: string, value: string) => void;
+	required?: boolean;
+	tooltip?: string;
+}
+
+export function SelectWithOption({
+	ariaLabel,
+	className,
+	disabled,
+	error,
+	feedbackMessage,
+	id,
+	items,
+	label,
+	onSelectChange,
+	required,
+	tooltip,
+	value,
+}: SelectWithOptionProps) {
+	const [dropdownActive, setDropdownActive] = useState(false);
+	const inputRef = useRef(null);
+
+	return (
+		<FieldBase
+			className={className}
+			disabled={disabled}
+			errorMessage={error}
+			helpMessage={feedbackMessage}
+			id={id}
+			label={label}
+			required={required}
+			tooltip={tooltip}
+		>
+			<>
+				<ClayIcon
+					className={classNames('base-select__input-icon', {
+						'base-select__input-icon--disabled': disabled,
+					})}
+					onClick={() =>
+						!disabled && setDropdownActive((active) => !active)
+					}
+					symbol="caret-double"
+				/>
+
+				<ClayAutocomplete.Input
+					aria-label={ariaLabel}
+					disabled={disabled}
+					onClick={() => setDropdownActive((active) => !active)}
+					placeholder={Liferay.Language.get('choose-an-option')}
```

Comment:
Hey @Manda-vit, there's a test failing related with your changes. In the "Then" section of the Action, the placeholder for the select is expected to be "Choose an Object" instead of "Choose an Option". You need to receive the value for placeholder as a parameter of the `SelectWithOption`, so you can pass the correct value on `ThenContainer` and you can keep `Liferay.Language.get('choose-an-option')` as the default value, in case no one passes other value for the placeholder.